sw/device/silicon_creator/lib/chip_info.h

```diff
@@ -0,0 +1,26 @@
+// Copyright lowRISC contributors.
+// Licensed under the Apache License, Version 2.0, see LICENSE for details.
+// SPDX-License-Identifier: Apache-2.0
+
+#ifndef OPENTITAN_SW_DEVICE_SILICON_CREATOR_LIB_CHIP_INFO_H_
+#define OPENTITAN_SW_DEVICE_SILICON_CREATOR_LIB_CHIP_INFO_H_
+
+#include <assert.h>
+#include <stdint.h>
+
+typedef struct chip_info {
+  uint32_t hw_rev;
+  uint64_t scm_revision;
+  char build_date_str[20];
```

Comment:
This string is feeling like more of a footgun to me after perusing [Embedded Systems: Struct Layouts for Long-term Storage](https://docs.google.com/document/d/12f9aQ5X_M2udJxHw8Q9jkWrg3zLb42ZZXqORajaa2x4/edit?usp=sharing). Maybe it should be an integral timestamp.

Reply:
I'm not against an integral timestamp, but why do you think this string is a footgun?  Is it not ASCIIZ?
It seems like the format of the string will be "YYYY-MM-DD HH:MM:SS" which is 19 characters of text, plus one for the `nul` terminator.